test.mjs

```diff
@@ -121,9 +122,9 @@ import {strict as assert} from 'assert'
   } catch (e) {
     console.log("This was expected", e)
   }
-  assert(processOutput.exitCode === 0)
-  assert(processOutput.stdout === '') // this is unexpected - intuitively would expect "Hello"
-  assert(processOutput.stderr === '')
+  if (processOutput) {
+    assert.fail('Expected failure, but got ' + processOutput[inspect.custom]())
```

Comment:
this fail-fast approach makes more sense to me - would have saved me from a bit of debugging